### DISCUSSION TITLE
Markdown-card: allow content from local files, so that 100% markdown content can be used

### DISCUSSION THREAD

**Discussion by pgorod** [home-assistant/feature-requests#615] [Dashboards & cards]
Hi. Currently the Markdown card only allows content that comes from the yaml of the Home Assistant configuration, things like:

```yaml
type: markdown
content: >
  ## My title

  Some text here
```

If I decide to include the content from a separate file, it will be like this:

```
content: !include markdown.yaml
```
And `markdown.yaml` will have to look like this:
```
>
  **Testing external file**.
```
### Problem: markdown has to make concessions for YAML!

In either if the two ways presented above, for longer or more complex markdown documents, those leading spaces are a killer. It's so easy to get things broken. In other words, the disadvantages of YAML are brought into the markdown world, needlessly.

And having a `markdown.yaml` file is not the same as having a true `markdown.ml` file: all external tools, generators, editors, validators, are going to stumble on the YAML bits.

The same goes for less technical human users. I can't let them edit the markdown because I am too afraid they'll mess up the two spaces indentation on the left, and break the HASS configuration altogether!

### Solution: allow proper inclusion of local files

This is  my feature request.

Allow for a configuration item to look like this:

```
type: markdown
content_include: my_markdown_file.md
```
Then just load the file and use it directly as content. No YAML indentation inside the file, just full markdown.

It either renders proper HTML, or produces a proper error from the markdown converter; but it doesn't break the entire HASS configuration because of basic indentation errors.

**Comment by pgorod** [comment]
This is still a good idea! And probably not very hard to implement....

**Comment by karwosts** [comment]
Frontend has no access to read arbitrary files in the config folder, so this would at least need some new core APIs to be possible.

**Comment by pgorod** [comment]
Couldn't the back-end send that "ready-made" to the front-end, I mean, send the `config` object with everything already in it?

**Reply by karwosts** [reply]
Not sure. I'll go ahead and move this to the main feature requests section, since it kind of sounds like this is isn't strictly going to be a frontend feature.
